app/src/main/java/com/duckduckgo/app/global/view/FireDialog.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.global.view
+
+import android.content.Context
+import android.support.design.widget.BottomSheetDialog
+import android.view.View
+import android.webkit.CookieManager
+import android.webkit.WebView
+import com.duckduckgo.app.browser.R
+import kotlinx.android.synthetic.main.sheet_fire_clear_data.*
+
+class FireDialog(context: Context, clearAllCallback: (() -> Unit)) : BottomSheetDialog(context) {
+
+    init {
+        val contentView = View.inflate(getContext(), R.layout.sheet_fire_clear_data, null)
+        setContentView(contentView)
+        clearAllOption.setOnClickListener {
+            clearCache()
+            clearCookies()
+            clearAllCallback()
+            dismiss()
+        }
+        cancelOption.setOnClickListener {
+            dismiss()
+        }
+    }
+
+    private fun clearCache() {
+        val webView = WebView(context)
+        webView.clearCache(true)
+        webView.clearHistory()
+    }
+
+    private fun clearCookies() {
+        CookieManager.getInstance().removeAllCookies(null)
```

Comment:
This method is async, so there is the slight risk that it would take a while to complete, and the user might visit a site before the cookie is completely removed. 

This seems like a small risk though, as I suspect however async it is, it will always beat the user. 

We could look to revisit this later and add a callback instead of null, and only inform the user all is done when the callback fires.

Thoughts?

Reply:
Makes sense, I'm making the change